src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+    }
+}
+
+f32 EnGo2_GetTargetXZSpeed(EnGo2* this) {
+    f32 yDist;
+    s32 index;
+
+    yDist = (this->actor.params & 0x1F) == GORON_DMT_BIGGORON ? 400.0f : 60.0f;
+
+    index = this->actor.params & 0x1F;
+
+    if (index == 1 && (fabsf(this->actor.yDistFromLink) < yDist) && (this->actor.xzDistFromLink < 400.0f)) {
+        return 9.0f;
+    } else {
+        return index == 0 ? 3.6000001f : 6.0f;
+    }
+}
+
+s32 EnGo2_IsCameraModified(EnGo2* this, GlobalContext* globalCtx) {
+    Camera* camera;
+
+    camera = globalCtx->cameraPtrs[0];
+    if ((this->actor.params & 0x1F) == GORON_DMT_BIGGORON) {
+        if (EnGo2_IsWakingUp(this)) {
+            Camera_ChangeSetting(camera, CAM_SET_TEPPEN);
+            func_8005AD1C(camera, 4);
+        } else if (!EnGo2_IsWakingUp(this) && (camera->setting == CAM_SET_TEPPEN)) {
+            Camera_ChangeSetting(camera, CAM_SET_DUNGEON1);
+            func_8005ACFC(camera, 4);
+        }
+    }
+
+    if ((this->actor.params & 0x1F) == GORON_FIRE_GENERIC || (this->actor.params & 0x1F) == GORON_CITY_ROLLING_BIG ||
+        (this->actor.params & 0x1F) == GORON_CITY_STAIRWELL || (this->actor.params & 0x1F) == GORON_DMT_BIGGORON ||
+        (this->actor.params & 0x1F) == GORON_MARKET_BAZAAR) {
+        return true;
+    } else if ((!CHECK_QUEST_ITEM(QUEST_MEDALLION_FIRE)) && (CHECK_OWNED_EQUIP(EQUIP_TUNIC, 1))) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void EnGo2_DefaultWakingUp(EnGo2* this) {
+    if (EnGo2_IsWakingUp(this)) {
+        this->unk_26E = 2;
+    } else {
+        this->unk_26E = 1;
+    }
+
+    if (this->unk_194.unk_00 != 0) {
+        this->unk_26E = 4;
+    }
+
+    this->isAwake = true;
+}
+
+void EnGo2_WakingUp(EnGo2* this) {
+    f32 xyzDist;
+    s32 isTrue;
+
+    if ((this->actor.params & 0x1F) == GORON_DMT_BIGGORON) {
+        xyzDist = 800.0f;
+    } else {
+        xyzDist = 200.0f;
+    }
+
+    xyzDist = SQ(xyzDist);
+    isTrue = true;
+
+    this->unk_26E = 1;
+
+    if ((this->actor.xyzDistFromLinkSq <= xyzDist) || (this->unk_194.unk_00 != 0)) {
+        this->unk_26E = 4;
+    }
+
+    this->isAwake = isTrue;
+}
+
+void EnGo2_BiggoronWakingUp(EnGo2* this) {
+    if (EnGo2_IsWakingUp(this) || this->unk_194.unk_00 != 0) {
+        this->unk_26E = 2;
+        this->isAwake = true;
+    } else {
+        this->unk_26E = 1;
+        this->isAwake = false;
+    }
+}
+
+void EnGo2_SelectGoronWakingUp(EnGo2* this) {
+    switch (this->actor.params & 0x1F) {
+        case GORON_DMT_BOMB_FLOWER:
+            this->isAwake = true;
+            this->unk_26E = EnGo2_IsWakingUp(this) ? 2 : 1;
+            break;
+        case GORON_FIRE_GENERIC:
+            EnGo2_WakingUp(this);
+            break;
+        case GORON_DMT_BIGGORON:
+            EnGo2_BiggoronWakingUp(this);
+            break;
+        case GORON_CITY_LINK:
+            if (!(CHECK_QUEST_ITEM(QUEST_MEDALLION_FIRE)) && CHECK_OWNED_EQUIP(EQUIP_TUNIC, 1)) {
```

Comment:
Oops, I just noticed I forgot to remove the parentheses here on the suggestion I made a few days ago
```suggestion
            if (!CHECK_QUEST_ITEM(QUEST_MEDALLION_FIRE) && CHECK_OWNED_EQUIP(EQUIP_TUNIC, 1)) {
```

Reply:
Great. There were two instances of this with the extra brackets